go/obscuronode/enclave/db/interfaces.go

```diff
@@ -42,37 +42,49 @@ type RollupResolver interface {
 	FetchRollupTxs(rollup *core.Rollup) (map[common.Hash]nodecommon.L2Tx, bool)
 	// StoreRollupTxs overwrites the transactions associated with a given rollup
 	StoreRollupTxs(rollup *core.Rollup, newTxs map[common.Hash]nodecommon.L2Tx)
+	// StoreGenesisRollup stores the rollup genesis
+	StoreGenesisRollup(rol *core.Rollup)
+	// FetchGenesisRollup returns the rollup genesis
+	FetchGenesisRollup() *core.Rollup
 }
 
-type StateStorage interface {
-	// FetchBlockState returns the state after ingesting the L1 Block with the given hash
-	FetchBlockState(hash obscurocommon.L1RootHash) (*BlockState, bool)
-	// FetchHeadState returns the state after ingesting the L1 Block at the head of the chain
+type BlockStateStorage interface {
+	// FetchBlockState returns the head rollup found in the block
+	FetchBlockState(blockHash obscurocommon.L1RootHash) (*BlockState, bool)
+	// FetchHeadState returns the head roolup
```

Comment:
@tudor-malene Typo - "roolup"

Reply:
thanks. Fixed in the current PR. (next iteration of this)